django_redis/pool.py

```diff
@@ -117,6 +121,53 @@ def get_connection_pool(self, params):
         return pool
 
 
+class SentinelConnectionFactory(ConnectionFactory):
+    def __init__(self, options):
+        # allow overriding the default SentinelConnectionPool class
+        options.setdefault(
+            "CONNECTION_POOL_CLASS", "redis.sentinel.SentinelConnectionPool"
+        )
+        super().__init__(options)
+
+        sentinels = options.get("SENTINELS")
+        if not sentinels:
+            raise ImproperlyConfigured(
+                "SENTINELS must be provided as a list of (host, port)."
+            )
+
+        # provide the connection pool kwargs to the sentinel in case it
+        # needs to use the socket options for the sentinels themselves
+        connection_kwargs = self.make_connection_params(None)
+        connection_kwargs.pop("url")
+        connection_kwargs.update(self.pool_cls_kwargs)
+        self._sentinel = Sentinel(
+            sentinels,
+            sentinel_kwargs=options.get("SENTINEL_KWARGS"),
+            **connection_kwargs,
+        )
```

Comment:
One thing I realized when reviewing https://github.com/jazzband/django-redis/pull/460/files#r556859461 for implications beyond "does the expected string get generated?" is that unlike the `ConnectionFactory` this will create a `Sentinel` object upon class creation. This _could_ be delayed to the first call to `get_connection_pool` if it was important not to construct a `Sentinel` object. It doesn't look like it would actually make a connection unless `single_connection_client` is `True`, but if py-redis changed its implementation this might make a connection when the factory is created rather when it's used.